broadway_api/daos/worker_node.py

```diff
@@ -19,19 +20,12 @@ def __init__(self, app):
             self._config["DB_PRIMARY"], WorkerNodeDao._COLLECTION
         )
 
-    def insert(self, obj):
```

Comment:
I do not think it would be a good design to remove the `insert` method and just the `update` method to add new documents too (using the `upsert` flag as you have). It just makes the code a little harder to understand.

Having a chain of if-elif-else would be more clean in my opinion:
```
if worker is None:
   dao.insert()
else:
   if worker.is_alive:
         abort()
   else:
         dao.update()
```

Reply:
yep